File: Backend/src/main/java/com/github/francisfire/anavis/models/DonationReport.java
package com.github.francisfire.anavis.models;

import java.util.Date;

import org.bson.types.Binary;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Document(collection = "donationreport")
public class DonationReport {

	@EqualsAndHashCode.Include
	@Id
	private String reportId;
	private byte[] reportFile;
	private String donorMail;
	private String officeMail;
	private Date date;

	public DonationReport(String donorMail, String officeMail, Date date) {
		this.donorMail = donorMail;
		this.officeMail = officeMail;
		this.date = date;
	}

}

File: Frontend/lib/models/donationreport.dart
class DonationReport {
  String _reportId;
  String _reportFile;
  String _donorMail;
  String _officeMail;
  String _date;
  DonationReport(String reportId, String reportFile, String donorMail,
      String officeMail, String date) {
    this._date = date;
    this._donorMail = donorMail;
    this._officeMail = officeMail;
    this._reportFile = reportFile;
    this._reportId = reportId;
  }

  String getReportId() {
    return _reportId;
  }

  String getReportFile() {
    return _reportFile;
  }

  String getDonorMail() {
    return _donorMail;
  }

  String getOfficeMail() {
    return _officeMail;
  }

  String getDate() {
    return _date;
  }
}
